template.yml.erb

```diff
@@ -693,6 +696,91 @@ Resources:
                 Period: 300
                 Stat: Sum
 <%end -%>
+# We use shortened versions of names for partition keys (eg, user_id),
+# but values will be a concatenation of the domain name and appropriate ID.
+# Values will look something like:
+# studio.code.org#123456 (user_requests table)
+# studio.code.org#UserId#123456 (blocked_users table)
```

Comment:
Just curious, why are we doing this concatenation instead of storing the domain separately? Is it so storage/lookup is quicker?

Reply:
Yeah good question, I was doing some reading on this as well. My understanding is DynamoDB is very fast when you are looking things up by the partition key. Our access pattern will always be to have a domain and user ID we want to look up, so we can get speedy results if we concatenate them and use the concatenated string to look up the record of interest.

Jessie had a couple of resources in her dev doc in case you want to learn more:
https://www.profit4cloud.nl/blog/designing-for-amazon-dynamodb/
https://aws.amazon.com/blogs/database/choosing-the-right-dynamodb-partition-key/